catroid/src/test/java/org/catrobat/catroid/test/content/project/ProjectMergeTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.project;
+
+import android.content.Context;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.test.MockUtil;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(JUnit4.class)
+public class ProjectMergeTest {
+	Project project1;
+	Project project2;
+
+	@Before
+	public void setUp() {
+		Context mockContext = MockUtil.mockContextForProject();
+		project1 = new Project(mockContext, "testProject");
+		project2 = new Project(mockContext, "testProject2");
+	}
+
+	@Test
+	public void testProjectGlobalVariableConflicts() {
+		project1.addUserVariable(new UserVariable("testVariable1", 1));
+		List<UserVariable> globalConflicts = ProjectManager.getInstance().getGlobalVariableConflicts(project1, project2);
+		assertEquals(globalConflicts.size(), 0);
+
+		project2.addUserVariable(new UserVariable("testVariable2", 1));
```

Comment:
Please split up your tests. Continuing to test after assertions makes a test hard to read.

For reference:
https://xp123.com/articles/3a-arrange-act-assert/